src/components/registration/steps.jsx

```diff
@@ -879,7 +880,11 @@ module.exports = {
             };
         },
         onNextStep: function () {
-            this.props.onNextStep();
+            this.props.onNextStep({
+                user: {
+                    username: this.props.studentUsername
```

Comment:
This seems convoluted - we're passing back the username that got passed in as a prop so that we can just pass it right back in to the next step. It would be more straightforward to pass the username we already have to both. It reminds me of the antipattern of setting state based on props - when you do that you are duplicating the source of truth: https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html